src/app/tools/page.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import ToolsPage from '@/components/tools/ToolsPage';
+import { Metadata } from 'next';
+
+export const metadata: Metadata = {
+  title: 'Browse Tools DTA',
+  description: 'DevToolsAcademy - Browse Tools',
+};
+interface SearchParams {
```

Comment:
code is functional and ok but we can add following improvements for future backlog:

1. let's extract SearchParams interface to `/types/pagination.ts`
2. let's move header JSX to separate component 
3. add proper page validation bounds

Otherwise LGTM 👍

Reply:
for example - 

// types.ts
export interface SearchParams {
  page?: string;
}

export interface PageProps {
  searchParams?: SearchParams;
}

// constants.ts
export const HEADER_CONFIG = {
  title: {
    main: "Browse Devtools for your",
    sub: "next product"
  },
  description: {
    main: "Discover new devtools from a well researched collection for hassle",
    sub: "free development of your next product"
  }
} as const;

// page.tsx
import { Metadata } from 'next';
import { PageProps } from './types';
import { HEADER_CONFIG } from './constants';
import { ToolsHeader } from './components/tools-header';
import { ToolsPageContent } from './components/tools-page-content';

export const metadata: Metadata = {
  title: 'Browse Tools DTA',
  description: 'DevToolsAcademy - Browse Tools',
  // Consider adding more metadata for SEO
  openGraph: {
    title: 'Browse Tools DTA',
    description: 'DevToolsAcademy - Browse Tools',
  },
};

export default function ToolsRoute({ searchParams }: PageProps) {
  const page = Math.max(1, Math.floor(Number(searchParams?.page) || 1));

  return (
    <main className="min-h-screen w-full">
      <ToolsHeader />
      <ToolsPageContent page={page} />
    </main>
  );
}

Reply:
// components/tools-header.tsx
import { cn } from "@/lib/utils";

export function ToolsHeader() {
  return (
    <div className="text-center pt-8">
      <h1 className={cn(
        "text-4xl sm:text-5xl md:text-5xl lg:text-6xl",
        "font-extrabold tracking-tight mb-4"
      )}>
        <GradientText>
          {HEADER_CONFIG.title.main}
          <span className="block">
            {HEADER_CONFIG.title.sub}
          </span>
        </GradientText>
      </h1>
      <p className={cn(
        "text-md sm:text-xl",
        "text-gray-100 text-opacity-50 mb-8"
      )}>
        {HEADER_CONFIG.description.main}{" "}
        <span className="md:block">
          {HEADER_CONFIG.description.sub}
        </span>
      </p>
    </div>
  );
}

// components/gradient-text.tsx
interface GradientTextProps {
  children: React.ReactNode;
  className?: string;
}

export function GradientText({ children, className }: GradientTextProps) {
  return (
    <span className={cn(
      "bg-gradient-to-b from-[#141414] to-white bg-clip-text text-transparent",
      className
    )}>
      {children}
    </span>
  );
}